### PR TITLE
fix: copy version tag related changes from importer to exporter

### LINKED ISSUE #22747: Bring importer 8.6/8.7 changes to exporter
### Description

After we copied importer business logic to exporter we continued adjusting importer when implementing bug fixes or new features.
We need to copy all this changes as a last step before we remove importer 8.6.

### TODO

Commits to migrate:

- [x] FNITransformer (double-check whether there are some meaningful changes): ccaf688 - all the changes are related with caches refactoring, which is not relevant for exporter. Nothinh to do here.
- [x] versionTag: b763d69, 53368f1, e77d2c4
- [ ] job and listeners related: 5686484, 916aef2, f92a942, 8c014ce, 1ff4ae3, 51a049b, 2250c4b
- [x] message subscription migration: 17689f4 (done within https://github.com/camunda/camunda/pull/25246)

### Additional Information

---------------------------------------------------------------------------------------------

### Acceptance Criteria

- [ ] Checked if the specific data needs to be provided for the QA process

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered
- [ ] Cross-team dependencies have been considered

### PR DESCRIPTION
**Pull request by sdorokhova**
PR contains some leftovers from commits https://github.com/camunda/camunda/commit/b763d697cc9c2cf9b7d9423c729702e7d7ae7fd4, https://github.com/camunda/camunda/commit/53368f14c0e9d8cf19a82e8f03be7055a82d177f, https://github.com/camunda/camunda/commit/e77d2c44856feac22f55f404fcf7c4510c15cd9d , that we not applied to exporter code earlier.

related with #22747

### CODE REVIEW

**Comment by sdorokhova** [comment]
Ran Operate integration tests on exporter: `ProcessCacheZeebeImportIT, ProcessZeebeImportIT and ListViewZeebeRecordProcessorIT`. All looks good.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandler.java [hunk 1]
```diff
@@ -217,6 +221,9 @@ protected String getProcessInstanceScript() {
         PROCESS_KEY,
         BPMN_PROCESS_ID,
         BPMN_PROCESS_ID,
+        PROCESS_VERSION_TAG,
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandler.java:224]: 🔧 the whole script can be simplified in the following format:
```java
return """
    if (ctx._source.%1$s == null || ctx._source.%1$s < params.%1$s) {
        ctx._source.%1$s = params.%1$s;
        ctx._source.%2$s = params.%2$s;
        ctx._source.%3$s = params.%3$s;
        ctx._source.%4$s = params.%4$s;
        ctx._source.%5$s = params.%5$s;
        if (params.%6$s != null) { ctx._source.%6$s = params.%6$s; }
        if (params.%7$s != null) { ctx._source.%7$s = params.%7$s; }
        if (params.%8$s != null) { ctx._source.%8$s = params.%8$s; }
        if (params.%9$s != null) { ctx._source.%9$s = params.%9$s; }
    }
    """
        .formatted(
            POSITION,
            PROCESS_NAME,
            PROCESS_VERSION,
            PROCESS_KEY,
            BPMN_PROCESS_ID,
            PROCESS_VERSION_TAG,
            START_DATE,
            END_DATE,
            STATE);
```

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ProcessHandler.java [hunk 2]
```diff
@@ -70,6 +70,7 @@ public void updateEntity(final Record<Process> record, final ProcessEntity entit
         .setKey(process.getProcessDefinitionKey())
         .setBpmnProcessId(process.getBpmnProcessId())
         .setVersion(process.getVersion())
+        .setVersionTag(process.getVersionTag())
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ProcessHandler.java:73]: 🔧 should we update the tests to assert this value too?

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandler.java [hunk 3]
```diff
@@ -152,6 +152,9 @@ public void flush(
     if (entity.getEndDate() != null) {
       updateFields.put(ListViewTemplate.END_DATE, entity.getEndDate());
     }
+    if (entity.getProcessVersionTag() != null) {
+      updateFields.put(ListViewTemplate.PROCESS_VERSION_TAG, entity.getProcessVersionTag());
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandler.java:156]: 🔧 should we increase unit test coverage to cover this field too?

**Review by mustafadagher — APPROVED** [review]
Looks good to me overall! I have only a couple of suggestions around unit test coverage, and the script.

Thanks Svetlana!

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandler.java (hunk 3 above)
Comment (sdorokhova) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandler.java:156]: Thanks, you're right! I extended the tests.

**Pull request merged**